### ISSUE TITLE
add better error explanations

### ISSUE THREAD

**Issue by iwa** [giorgi-o/SkinPeek#104]
lots of time when the bot returns an error, it can often be fixed by a relogin
couldn't you add some better errors messages so user aren't confused by whats happening?

**Comment by giorgi-o** [comment]
I agree, however the issue is that the error changes depending on what is causing the current session to not work, whether it's the Cloudflare firewall, the authentication, the shop server, etc.

Could you send me a screenshot of a console error that can be fixed by relogging? If so I can change that error to tell the user they have to /login again.

**Comment by iwa** [comment]
problem is i don't even have logs about the error...
see below

![Capture d'écran 2024-05-10 004435](https://github.com/giorgi-o/SkinPeek/assets/19956672/8e26c15b-99f8-4df5-9a82-8ea2c0f64b67)
![Capture d'écran 2024-05-10 004514](https://github.com/giorgi-o/SkinPeek/assets/19956672/d5cfbc02-6299-4601-9cf9-d6106d9359af)

**Comment by iwa** [comment]
i think this would be the issue?

![image](https://github.com/giorgi-o/SkinPeek/assets/19956672/a4705947-bd6c-426e-847e-05595b57a6cd)

problem is i currently selfhost the bot on my homelab, so i don't get why we would get cf error?

**Comment by giorgi-o** [comment]
The way the Cloudflare firewall works isn't on/off, but it calculates a "score" based on many factors, to determine whether the username/password login is coming from an official riot client or not.

The IP address plays a part (whether it's an enterprise or home address, how many different accounts share that IP, etc) but also tons of other metrics like user-agent, [TLS fingerprinting](https://fingerprint.com/blog/what-is-tls-fingerprinting-transport-layer-security/), even stuff like the timing of the response. There are world experts at Cloudflare whose entire job it is to maintain this algorithm btw.

SkinPeek tries its best to pretend to be the Rito launcher, but it's not 100%. And if Cloudflare detects you, there's not much I or anyone can do unfortunately, sorry.
